pkg/ottl/ottlfuncs/func_trim.go

```diff
@@ -0,0 +1,55 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type TrimArguments[K any] struct {
+	Target ottl.GetSetter[K]
```

Comment:
This function should be a `Converter` so it should return the modified value instead of setting.  This should be a `StringGetter`